slice bounds out of range in pfilter

```
09:51:31 INFO: syncthing v1.17.0-rc.1 "Fermium Flea" (go1.16.3 linux-amd64) docker@build.syncthing.net 2021-05-11 06:02:19 UTC [noupgrade]
Panic at 2021-05-11T12:51:28Z
panic: runtime error: slice bounds out of range [:-1]

goroutine 77 [running]:
github.com/AudriusButkevicius/pfilter.(*PacketFilter).readMsgUdp(0xc0000cd4d0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/AudriusButkevicius/pfilter@v0.0.0-20210510194644-fad42c10c5ac/filter.go:173 +0x276
github.com/AudriusButkevicius/pfilter.(*PacketFilter).loop(0xc0000cd4d0, 0xc00023ef40)
	github.com/AudriusButkevicius/pfilter@v0.0.0-20210510194644-fad42c10c5ac/filter.go:180 +0x3f
created by github.com/AudriusButkevicius/pfilter.(*PacketFilter).Start
	github.com/AudriusButkevicius/pfilter@v0.0.0-20210510194644-fad42c10c5ac/filter.go:146 +0x7b
```

https://sentry.syncthing.net/syncthing/syncthing/issues/11037/?query=version%3Av1.17.0-rc.1%20is%3Aunresolved

(Dang, released rc.2 too quickly)

--- a/go.mod
+++ b/go.mod
@@ -1,7 +1,7 @@
 module github.com/syncthing/syncthing
 
 require (
-	github.com/AudriusButkevicius/pfilter v0.0.0-20210510194644-fad42c10c5ac
+	github.com/AudriusButkevicius/pfilter v0.0.0-20210511165305-e9aaf99ab213
 	github.com/AudriusButkevicius/recli v0.0.5
 	github.com/alecthomas/kong v0.2.16
 	github.com/bkaradzic/go-lz4 v0.0.0-20160924222819-7224d8d8f27e

